### ISSUE TITLE
Will remacs attempt to untangle the years of rendering emulations?

### ISSUE THREAD

**Issue by socketwiz** [Wilfred/remacs#700]
tldr; This is not a bug, more of a question/discussion

I ran across this very informative article on the history of Emacs rendering:
https://www.facebook.com/notes/daniel-colascione/buttery-smooth-emacs/10155313440066102/

I was just curious if remacs would attempt to make this better or is it really a just trying to replace the C parts with Rust?

**Comment by shaleh** [comment]
Right now, replace the C. Long term, yeah warts can go away. Better (or at least better documented) internals for characters, useful threading, etc.

But until the C goes away we are chasing too many things.

**Comment by shaleh** [comment]
We no longer need to worry about DOS or OS/2 for instance. This helps simplify things.

**Comment by shanavas786** [comment]
Article was very informative

> One day, a fool wanted to run Emacs in a GUI as a native GUI program. The rest is ChangeLog.

This made my day LOL

**Comment by TinaRussell** [comment]
Yeah, I hope we can get better GUI rendering in the future of Remacs, though I can understand that the ship-of-Theseus task of replacing the C code comes first. I’m so tired of forum threads where people ask about if Emacs can display video/render a 3D object/other modern stuff, and even when most replies are like “no, sorry,” or “well there’s this impressive kludge somebody wrote, try that,” both genuine attempts to be helpful, there’s always one or two guys like “WHY ARE YOU TRYING TO DO THAT Emacs is about TEXT TEXT TEXT any modern feature will have to rewrite 30-year-old C code and NOBODY WANTS TO TOUCH THAT blah blah blah go crying home to Visual Studio you script-snorting pleb"* and it just makes my heart sink. Anyway, bless you guys for not giving up and choosing to go head-first into all that cruft, it is a toil worthy of Hephaistos himself.

* This is a slight exaggeration, though it’s how those replies always come across to me. Computing belongs to the realm of my inner child, and she may have some serious issues

**Comment by shaleh** [comment]
I feel the same and that is why I hack on this.
